Source/WebCore/page/Navigation.cpp

```diff
@@ -93,12 +93,12 @@ bool Navigation::canGoForward() const
 }
 
 // https://html.spec.whatwg.org/multipage/nav-history-apis.html#getting-the-navigation-api-entry-index
-static std::optional<size_t> getEntryIndexOfHistoryItem(const Vector<Ref<NavigationHistoryEntry>>& entries, const HistoryItem& item)
+static std::optional<size_t> getEntryIndexOfHistoryItem(const Vector<Ref<NavigationHistoryEntry>>& entries, const HistoryItem& item, size_t start = 0)
 {
     size_t index = 0;
     // FIXME: We could have a more efficient solution than iterating through a list.
     for (auto& entry : entries) {
-        if (entry->associatedHistoryItem() == item)
+        if (entry->associatedHistoryItem() == item && index >= start)
```

Comment:
Instead of adding this constraint and unnecessarily iterating entries before the start just to do a comparison and increment an integer, let's start at start and only do a comparison on the possibilities.  No c++11-style iteration in this case.

Alternatively, we could pass in a std::span of Refs and have the caller do a subspan instead of passing in a vector and start.

Reply:
I rewrote to use a subspan internally. When trying to hand a suitable span as parameter the index counting did not take the skipped entries into account so the calculated m_currentEntryIndex was off.